vars/piperPipelineStageInit.groovy

```diff
@@ -106,7 +116,21 @@ void call(Map parameters = [:]) {
     def stageName = StageNameProvider.instance.getStageName(script, parameters, this)
 
     piperStageWrapper (script: script, stageName: stageName, stashContent: [], ordinal: 1, telemetryDisabled: true) {
-        def scmInfo = checkout(parameters.checkoutMap ?: scm)
+        def skipCheckout = parameters.skipCheckout
+        if (skipCheckout != null && !(skipCheckout instanceof Boolean)) {
+            error "[${STEP_NAME}] Parameter skipCheckout has to be of type boolean. Instead got '${skipCheckout.getClass()}'"
```

Comment:
[minor] `getClass().getName` ?

Reply:
Actually this reminded me that getClass() is anyway not good in case somebody provides us a map here, because then the **key** 'class' will be retrieved by the generated groovy-getter and this is either some value or `null`, but not the class. Fixed.